nutils/solver.py

```diff
@@ -947,15 +949,20 @@ def _integrate_blocks(*blocks, arguments, mask):
   assert not list(data)
   return nrg + [sparse.toarray(sparse.block(res)), matrix.fromsparse(sparse.block(jac), inplace=True)]
 
-def _argshapes(integrals):
-  '''merge argshapes of multiple integrals'''
-
-  argshapes = {}
-  for target, shape in (item for integral in integrals if integral for item in integral.argshapes.items()):
-    if target not in argshapes:
-      argshapes[target] = shape
-    elif argshapes[target] != shape:
-      raise ValueError('shapes do not match for target {!r}: {} != {}'.format(target, argshapes[target], shape))
-  return argshapes
+@functools.lru_cache(128)
```

Comment:
That's because the `Integral.argshapes` property was cached as well. But indeed, `Evaluable.arguments` is cached, so it makes sense to remove the cache.